src/integration_tests/helpers/integration_helpers.py

```diff
@@ -0,0 +1,229 @@
+import requests
+from config.config_factory import ConfigFactory
+from firebase_admin import _apps, firestore
+from google.cloud import storage
+from requests.adapters import HTTPAdapter
+from urllib3 import Retry
+
+config = ConfigFactory.get_config()
+
+if not _apps:
+    firebase_admin.initialize_app()
+
+db = firestore.client()
+
+
+def setup_session() -> requests.Session:
+    """
+    Method to setup a http/s session to facilitate testing.
+
+    Parameters:
+        None
+
+    Returns:
+        Session: a http/s session.
+    """
+    session = requests.Session()
+    retry = Retry(connect=3, backoff_factor=0.5)
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount("http://", adapter)
+    session.mount("https://", adapter)
+
+    return session
+
+
+def generate_headers() -> dict[str, str]:
+    """
+    Method to create headers for authentication if connecting to a remote version of the API.
+
+    Parameters:
+        None
+
+    Returns:
+        dict[str, str]: the headers required for remote authentication.
+    """
+    headers = {}
+    if not config.API_URL.__contains__("localhost"):
+        try:
+            auth_req = google.auth.transport.requests.Request()
+            auth_token = google.oauth2.id_token.fetch_id_token(auth_req, config.API_URL)
+        except Exception:
+            auth_token = os.environ.get("ACCESS_TOKEN")
+
+        headers = {"Authorization": f"Bearer {auth_token}"}
+
+    return headers
+
+
+def load_json(filepath: str) -> dict:
+    """
+    Method to load json from a file.
+
+    Parameters:
+        filepath: string specifiing the location of the file to be loaded.
+
+    Returns:
+        dict: the json object from the specifiede file.
+    """
+    with open(filepath) as f:
+        return json.load(f)
+
+
+def create_dataset(
+    filename: str, dataset: dict, session: requests.Session, headers: dict[str, str]
+) -> int | None:
+    """
+    Method to create a dataset using either the remote new dataset function or the local version.
+
+    Parameters:
+        filename: the filename to use for the dataset
+        dataset: the dataset to be created
+        session: a session instance for http/s connections
+        headers: the relevant headers for authentication for http/s calls
+
+    Returns:
+        int | None: status code for local function and no return for remote.
+    """
+    if config.API_URL.__contains__("local"):
+        simulate_post_dataset_request = session.post(
+            "http://localhost:3006", json=dataset
+        )
+        return simulate_post_dataset_request.status_code
+    else:
+        bucket = storage.Client().bucket(config.DATASET_BUCKET_NAME)
+        blob = bucket.blob(filename)
+        blob.upload_from_string(
+            json.dumps(dataset, indent=2), content_type="application/json"
+        )
+
+        wait_until_dataset_ready(
+            dataset["survey_id"], dataset["period_id"], session, headers
+        )
```

Comment:
Worth raising an error here if this doesn't work instead of returning 'None'?

Reply:
Interestingly it already does as that method throws an exception if the dataset is not found